### ISSUE TITLE
pylint error invalid-name causes an error for one variable that has a different version that is not a constant

### ISSUE THREAD

**Issue by ollie-iterators** [PyCQA/pylint#8212]
### Bug description

invalid-name causes an error for a variable that has a different value that is not a constant variable

### Command used

```shell
def leap_year(year):
    """ Function used to determine whether a given year is a leap year """
    if year % 400 == 0:
        return (True, year)
    if year % 100 == 0:
        return (False, year)
    if year % 4 == 0:
        return (True, year)
    else:
        return (False, year)

TEST_CODE = True
if TEST_CODE:
    theDecode = leap_year(2020)
else:
    theDecode = False
```

### Pylint output

```shell
Constant name "theDecode" doesn't conform to UPPER_CASE naming stylePylint(C0103:invalid-name)
```

### Expected behavior

There is no error for the variable "theDecode" because it is not just a constant variable.

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.11.1 (v3.11.1:a7a450f84a, Dec  6 2022, 15:24:06) [Clang 13.0.0 (clang-1300.0.29.30)]
```

### Additional dependencies

pylint 2.16.1
astroid 2.14.1
Python 3.11.1 (v3.11.1:a7a450f84a, Dec  6 2022, 15:24:06) [Clang 13.0.0 (clang-1300.0.29.30)]
